test/CadlRanchProjectsNonAzure.Tests/type-property-optionality.cs

```diff
@@ -0,0 +1,480 @@
+            Assert.AreEqual(true, BooleanLiteralProperty.FromResponse(response.GetRawResponse()).Property);
+        });
+
+        [Test]
+        public Task Type_Property_Optional_BooleanLiteral_getDefault() => Test(async (host) =>
+        {
+            var response = await new OptionalClient(host, null).GetBooleanLiteralClient().GetDefaultAsync();
+            Assert.AreEqual(null, BooleanLiteralProperty.FromResponse(response.GetRawResponse()).Property);
+        });
+
+        [Test]
+        public Task Type_Property_Optional_BooleanLiteral_putAll() => Test(async (host) =>
+        {
+            BooleanLiteralProperty data = new()
+            {
+                Property = true
+            };
+            var response = await new OptionalClient(host, null).GetBooleanLiteralClient().PutAllAsync(data);
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_Optional_BooleanLiteral_putDefault() => Test(async (host) =>
+        {
+            var response = await new OptionalClient(host, null).GetBooleanLiteralClient().PutDefaultAsync(new BooleanLiteralProperty());
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_Optional_String_getAll() => Test(async (host) =>
+        {
+            var response = await new OptionalClient(host, null).GetStringClient().GetAllAsync();
+            Assert.AreEqual("hello", StringProperty.FromResponse(response.GetRawResponse()).Property);
+        });
+
+        [Test]
+        public Task Type_Property_Optional_String_getDefault() => Test(async (host) =>
+        {
+            var response = await new OptionalClient(host, null).GetStringClient().GetDefaultAsync();
+            Assert.AreEqual(null, StringProperty.FromResponse(response.GetRawResponse()).Property);
+        });
+
+        [Test]
+        public Task Type_Property_Optional_String_putAll() => Test(async (host) =>
+        {
+            StringProperty data = new()
+            {
+                Property = "hello"
+            };
+            var  response = await new OptionalClient(host, null).GetStringClient().PutAllAsync(data);
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_Optional_String_putDefault() => Test(async (host) =>
+        {
+            var response = await new OptionalClient(host, null).GetStringClient().PutDefaultAsync(new StringProperty());
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_Optional_Bytes_getAll() => Test(async (host) =>
+        {
+            var response = await new OptionalClient(host, null).GetBytesClient().GetAllAsync();
+            BinaryDataAssert.AreEqual(BinaryData.FromString("hello, world!"), BytesProperty.FromResponse(response.GetRawResponse()).Property);
```

Comment:
same here, we could just use:
```suggestion
            BinaryDataAssert.AreEqual(BinaryData.FromString("hello, world!"), response.Value.Property);
```

Reply:
fixed